libs/ledger/src/consensus/consensus.cpp

```diff
@@ -610,6 +627,17 @@ NextBlockPtr Consensus::GenerateNextBlock()
     notarisation_->NotariseBlock(*ret);
   }
 
+#ifndef NDEBUG
+  try
+  {
+    GetBeginningOfAeon(*ret, chain_);
```

Comment:
Is this supposed to be kept here?

Reply:
See, earlier in the code, the weight is already calculated via `GetBlockGenerationWeight()`, so the beginning of aeon should not fail at this moment. Consequently, I've inserted this block as a debug-time assertion for that "should not".